core/src/main/java/org/mule/runtime/core/transformer/types/DataTypeFactory.java

```diff
@@ -308,4 +306,52 @@ else if (classInMap != null)
         proxyClassCache.put(typeName, new ProxyIndicator(type, isProxy));
         return isProxy;
     }
+
+    public static <T> DataType<T> createFromDataType(DataType<T> typeFrom, String mimeType, String encoding)
```

Comment:
Are these methods temporary, until proper builder is defined? If so, add TODO's.